code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,395 @@
+			return
+	var/datum/mind/M = user.mind
+	var/datum/antagonist/ecult/cultie = M.has_antag_datum(/datum/antagonist/ecult)
+
+	for(var/X in cultie.get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		EK.mansus_grasp_act(target, user, proximity_flag, click_parameters)
+
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.adjustBruteLoss(10)
+		C.AdjustKnockdown(1 SECONDS)
+		return
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion
+	name = "Aggressive Spread"
+	desc = "Spreads rust onto nearby turfs."
+
+	school = "transmutation"
+	charge_max = 300 //twice as long as mansus grasp
+	clothes_req = FALSE
+	invocation = "A'GRSV SPR'D"
+	invocation_type = "whisper"
+	range = 3
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava,/turf/open/chasm,/turf/open/floor/plating/rust))
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/cast(list/targets, mob/user = usr)
+	playsound(get_turf(user), 'sound/items/welder.ogg', 75, TRUE)
+	for(var/turf/T in targets)
+		///What we want is the 3 tiles around the user and the tile under him to be rusted, so min(dist,1)-1 causes us to get 0 for these tiles, rest of the tiles are based on chance
+		var/chance = 100 - (max(get_dist(T,user),1)-1)*100/(range+1)
+		if(!prob(chance))
+			continue
+		if(!is_type_in_typecache(T, blacklisted_turfs))
+			T.ChangeTurf(/turf/open/floor/plating/rust)
+			continue
+		if(T.type == /turf/closed/wall/rust)
+			T.ScrapeAway()
+			continue
+		if(T.type == /turf/closed/wall/r_wall/rust && prob(50))
+			T.ScrapeAway()
+			continue
+		if(T.type == /turf/closed/wall)
+			T.ChangeTurf(/turf/closed/wall/rust)
+			continue
+		if(T.type == /turf/closed/wall/r_wall && prob(50))
+			T.ChangeTurf(/turf/closed/wall/r_wall/rust)
+			continue
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small
+	name = "Rust Conversion"
+	desc = "Spreads rust onto nearby turfs."
+	range = 2
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/big
+	name = "Rustbringer's rite"
+	desc = "Spreads rust onto a lot of turfs, and damages items worn on other peoplke"
+	range = 8
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/big/cast(list/targets, mob/user)
+	. = ..()
+	for(var/turf/T in targets)
+
+		for(var/Y in T.GetAllContents())
+
+			if(!is_type_in_list(/mob/living/carbon,T.GetAllContents()))
+				break
+
+			if(!istype(Y,/mob/living/carbon))
+				continue
+
+			var/mob/living/carbon/C = Y
+			if(C == user)
+				continue
+
+			for(var/X in C.GetAllContents())
+				if(istype(X,/obj/item))
+					var/obj/item/I = X
+					if(prob(75))
+						I.take_damage(rand(50,500))
+
+/obj/effect/proc_holder/spell/targeted/touch/ash_leech
+	name = "Blood Siphon"
+	desc = "Touch spell that heals you while damaging the enemy."
```

Comment:
kind of overlaps with the hand of mansus spell, no?

Reply:
i decided to keep them seperate because if you chose flesh and clicked on a dead person you would instantly kill them because this also deals damage

Reply:
speaking of that, have you ensured that the damage/knockdown from the touch of mansus spell will come BEFORE the riders from that spell? otherwise, you'll deal 10 damage to each of your new ghouls, no? also, it'd be nice if the knockdown and such occurred BEFORE the yeeting from the ash cultist rider effect

Reply:
done